client/client.go

```diff
@@ -38,32 +38,47 @@ type DataExportClient interface {
 
 // Client represents a HTTP client for making requests to the FullStory API.
 type Client struct {
-	HTTPClient *http.Client
-	Config     *config.Config
+	HTTPClient    *http.Client
+	Configuration *config.Config
 }
 
 var _ DataExportClient = (*Client)(nil)
 
-// NewClient returns a Client initialized with http.DefaultClient and the
-// supplied apiToken.
-func NewClient(config *config.Config) *Client {
+// FullStoryAPIKeyRoundTripper uses the http.RoundTripper interface to provide an
+// implementation of RoundTripper that adds the special header(s) for use with the
+// full story export mechanism.
+type FullStoryAPIKeyRoundTripper struct {
+	cfg *config.Config
+}
+func (f *FullStoryAPIKeyRoundTripper) RoundTrip(req*http.Request) (*http.Response,error){
+	f.addAuthHeaders(req)
+	return http.DefaultClient.Transport.RoundTrip(req)
+}
+func (f *FullStoryAPIKeyRoundTripper) addAuthHeaders(req *http.Request) {
+	req.Header.Set("Authorization", "Basic "+f.cfg.FsApiToken)
+	for _, header:=range f.cfg.AdditionalHttpHeader{
+		req.Header.Add(header.Key, header.Value)
+	}
+}
+
+func NewClient(cfg *config.Config) *Client {
```

Comment:
I don't think you need any changes to this file if you rebase off of master.

Reply:
Yep... great minds...